src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -1225,77 +1210,73 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
-						await this._updateNeverProblemMatcherSetting(selected.setting);
-						return task;
-					} else {
-						return task;
-					}
-				} else {
-					return undefined;
-				}
-			});
+		});
+		entries.unshift({ type: 'separator', label: nls.localize('TaskService.associate', 'associate') });
+		let taskType: string;
+		if (CustomTask.is(task)) {
+			const configProperties: TaskConfig.IConfigurationProperties = task._source.config.element;
+			taskType = (<any>configProperties).type;
+		} else {
+			taskType = task.getDefinition().type;
+		}
+		entries.unshift(
+			{ label: nls.localize('TaskService.attachProblemMatcher.continueWithout', 'Continue without scanning the task output'), matcher: undefined },
+			{ label: nls.localize('TaskService.attachProblemMatcher.never', 'Never scan the task output for this task'), matcher: undefined, never: true },
+			{ label: nls.localize('TaskService.attachProblemMatcher.neverType', 'Never scan the task output for {0} tasks', taskType), matcher: undefined, setting: taskType },
+			{ label: nls.localize('TaskService.attachProblemMatcher.learnMoreAbout', 'Learn more about scanning the task output'), matcher: undefined, learnMore: true }
+		);
+		const problemMatcher = await this._quickInputService.pick(entries, { placeHolder: nls.localize('selectProblemMatcher', 'Select for which kind of errors and warnings to scan the task output') });
+		if (!problemMatcher) {
+			return task;
+		}
+		if (problemMatcher.learnMore) {
+			this._openDocumentation();
+			return undefined;
+		}
+		if (problemMatcher.never) {
+			this.customize(task, { problemMatcher: [] }, true);
+			return task;
+		}
+		if (problemMatcher.matcher) {
+			const newTask = task.clone();
+			const matcherReference = `$${problemMatcher.matcher.name}`;
+			const properties: ICustomizationProperties = { problemMatcher: [matcherReference] };
+			newTask.configurationProperties.problemMatchers = [matcherReference];
+			const matcher = ProblemMatcherRegistry.get(problemMatcher.matcher.name);
+			if (matcher && matcher.watching !== undefined) {
+				properties.isBackground = true;
+				newTask.configurationProperties.isBackground = true;
+			}
+			this.customize(task, properties, true);
+			return newTask;
+		}
+		if (problemMatcher.setting) {
+			await this._updateNeverProblemMatcherSetting(problemMatcher.setting);
 		}
-		return Promise.resolve(task);
+		return task;
 	}
 
-	private _getTasksForGroup(group: TaskGroup): Promise<Task[]> {
-		return this._getGroupedTasks().then((groups) => {
-			const result: Task[] = [];
-			groups.forEach((tasks) => {
-				for (const task of tasks) {
-					const configTaskGroup = TaskGroup.from(task.configurationProperties.group);
-					if (configTaskGroup?._id === group._id) {
-						result.push(task);
-					}
+	private async _getTasksForGroup(group: TaskGroup): Promise<Task[]> {
+		const groups = await this._getGroupedTasks();
+		const result: Task[] = [];
+		for (const tasks of Object.entries(groups)) {
```

Comment:
Does forEach of a map in typescript iterate through values instead of entries?

JavaScript goes through entries https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach

Reply:
I've used object.entries elsewhere and believe it just goes through each key,value

Reply:
Hm, maybe `Map.prototype.entries()` instead of `Object.entries()`?

`Map.prototype.forEach` does iterate through entries, but it looks like the original code drops the entry key (which is the second arg to the callback):

```ts
		groups.forEach((value, folder) => {
			// type of value is Task[]
			// type of folder is string
		});
```

I'm not sure about `Object.entries`. TypeScript says that the type of `tasks` is `[string, any]`:
```ts
		for (const tasks of Object.entries(groups)) {
			// type of tasks is [string, any]
			for (const task of tasks) { // type of task is `any`
```

and also I don't see the map entries being printed if I loop over `Object.entries` in the MDN playground:

![image](https://user-images.githubusercontent.com/30305945/182288060-6741562c-7f83-4cc4-95ff-c5f31c3734ee.png)

Reply:
Maybe you'd need json.stringify to print it because an entry is an object?

Reply:
Looks like an empty array: 
![image](https://user-images.githubusercontent.com/30305945/182289967-f3255084-29ff-407d-b703-596dc985d37f.png)